conformance/05_referrers.go

```diff
@@ -0,0 +1,276 @@
+				resp, err := client.Do(req)
+				Expect(err).To(BeNil())
+				Expect(resp.StatusCode()).To(SatisfyAll(
+					BeNumerically(">=", 200),
+					BeNumerically("<", 300)))
+			})
+
+			g.Specify("Populate registry with test references manifest (ArtifactType, config.MediaType = scratch)", func() {
+				SkipIfDisabled(referrers)
+				RunOnlyIf(runReferencesSetup)
+				req := client.NewRequest(reggie.PUT, "/v2/<name>/manifests/<reference>",
+					reggie.WithReference(refsManifestLayerArtifactDigest)).
+					SetHeader("Content-Type", "application/vnd.oci.image.manifest.v1+json").
+					SetBody(refsManifestLayerArtifactContent)
+				resp, err := client.Do(req)
+				Expect(err).To(BeNil())
+				Expect(resp.StatusCode()).To(SatisfyAll(
+					BeNumerically(">=", 200),
+					BeNumerically("<", 300)))
+			})
+		})
+
+		g.Context("Get references", func() {
+			g.Specify("GET request to nonexistent blob should result in empty 200 response", func() {
+				SkipIfDisabled(referrers)
+				req := client.NewRequest(reggie.GET, "/v2/<name>/referrers/<digest>",
+					reggie.WithDigest(dummyDigest))
+				resp, err := client.Do(req)
+				Expect(err).To(BeNil())
+				Expect(resp.StatusCode()).To(Equal(http.StatusOK))
+
+				var index Index
+				err = json.Unmarshal(resp.Body(), &index)
+				Expect(err).To(BeNil())
+				Expect(len(index.Manifests)).To(BeZero())
+			})
+
+			g.Specify("GET request to existing blob should yield 200", func() {
+				SkipIfDisabled(referrers)
+				req := client.NewRequest(reggie.GET, "/v2/<name>/referrers/<digest>",
+					reggie.WithDigest(manifests[4].Digest))
+				resp, err := client.Do(req)
+				Expect(err).To(BeNil())
+				Expect(resp.StatusCode()).To(Equal(http.StatusOK))
+				if h := resp.Header().Get("Docker-Content-Digest"); h != "" {
+					Expect(h).To(Equal(configs[4].Digest))
+				}
+
+				var index Index
+				err = json.Unmarshal(resp.Body(), &index)
+				Expect(err).To(BeNil())
+				Expect(len(index.Manifests)).ToNot(BeZero())
+			})
+
+			g.Specify("GET request to existing blob with filter should yield 200", func() {
+				SkipIfDisabled(referrers)
+				req := client.NewRequest(reggie.GET, "/v2/<name>/referrers/<digest>",
+					reggie.WithDigest(manifests[4].Digest)).
+					SetQueryParam("artifactType", "application/vnd.oci.conformance")
+				resp, err := client.Do(req)
+				Expect(err).To(BeNil())
+				Expect(resp.StatusCode()).To(Equal(http.StatusOK))
+				if h := resp.Header().Get("Docker-Content-Digest"); h != "" {
+					Expect(h).To(Equal(configs[4].Digest))
+				}
+
+				var index Index
+				err = json.Unmarshal(resp.Body(), &index)
+				Expect(err).To(BeNil())
+				Expect(len(index.Manifests)).ToNot(BeZero())
+
+				// also check resp header "OCI-Filters-Applied: artifactType" denoting that an artifactType filter was applied
+				Expect(resp.Header().Get("OCI-Filters-Applied")).ToNot(BeEmpty())
+				Expect(resp.Header().Get("OCI-Filters-Applied")).To(Equal("application/vnd.oci.conformance"))
```

Comment:
Filter support is a SHOULD, so at most we can warn. I'd prefer we push a third referrer with a different type, and then conditionally handle the response based on if the registry returns the `OCI-Filters-Applied` header. If it's missing, all 3 referrers should be returned. Otherwise if the header is defined with the requested value, we should verify the two requested referrers were returned and not the 3rd.

Reply:
```
      WARN: you have failed this test, but ignoring
      Expected
         <string>: application/vnd.nhl.peanut.butter.bagel
      to equal
         <string>:
   ```
   
   ^ warn support added

Reply:
Modified implementation.